egograph/backend/tests/unit/test_dataset_catalog.py

```
@@ -134,7 +136,63 @@ def test_duplicate_dataset_id_fails_fast():
         path="test/b",
         partition_policy=PartitionPolicy.MONTHLY,
         compaction_strategy=CompactionStrategy.NONE,
+        required_columns=("id",),
     )
 
     with pytest.raises(ValueError, match="duplicate_dataset_id: duplicate.dataset"):
```

Ruff (0.16.0)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)